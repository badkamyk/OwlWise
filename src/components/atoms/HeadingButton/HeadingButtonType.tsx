export type HeadingButtonType = {
	text: string;
	href?: string;
	type?: "button" | "submit" | "reset";
	disabled?: boolean;
};
