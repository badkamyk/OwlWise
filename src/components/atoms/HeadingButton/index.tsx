import { HeadingButtonType } from "./HeadingButtonType";

export default function HeadingButton({
	text,
	href,
	type,
	disabled,
}: HeadingButtonType) {
	return (
		<button
			data-testid="heading-button"
			type={type}
			disabled={disabled}
			className="group relative mb-2 mr-2 inline-flex items-center justify-center overflow-hidden rounded-lg bg-gradient-to-br from-green-400 to-blue-600 p-0.5 text-sm font-medium text-gray-900 hover:text-white focus:outline-none focus:ring-4 focus:ring-green-200 group-hover:from-green-400 group-hover:to-blue-600 dark:text-white dark:focus:ring-green-800"
		>
			<a
				href={href}
				className="relative rounded-md bg-white px-5 py-2.5 transition-all duration-75 ease-in group-hover:bg-opacity-0 dark:bg-gray-900"
			>
				{text}
			</a>
		</button>
	);
}
