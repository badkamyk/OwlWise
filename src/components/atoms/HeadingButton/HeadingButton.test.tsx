import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import HeadingButton from "./index";

describe("HeadingButton", () => {
	const result = render(<HeadingButton href="Home" text={"Home"} />);
	it("renders a link with href and classes", () => {
		expect(result).toBeTruthy();
		expect(screen.getByText("Home")).toBeTruthy();
		expect(result).toMatchSnapshot();
	});
});
