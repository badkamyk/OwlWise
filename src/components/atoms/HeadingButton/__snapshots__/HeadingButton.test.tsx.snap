// Vitest Snapshot v1

exports[`HeadingButton > renders a link with href and classes 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="group relative mb-2 mr-2 inline-flex items-center justify-center overflow-hidden rounded-lg bg-gradient-to-br from-green-400 to-blue-600 p-0.5 text-sm font-medium text-gray-900 hover:text-white focus:outline-none focus:ring-4 focus:ring-green-200 group-hover:from-green-400 group-hover:to-blue-600 dark:text-white dark:focus:ring-green-800"
      >
        <a
          class="relative rounded-md bg-white px-5 py-2.5 transition-all duration-75 ease-in group-hover:bg-opacity-0 dark:bg-gray-900"
          href="Home"
        >
          Home
        </a>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="group relative mb-2 mr-2 inline-flex items-center justify-center overflow-hidden rounded-lg bg-gradient-to-br from-green-400 to-blue-600 p-0.5 text-sm font-medium text-gray-900 hover:text-white focus:outline-none focus:ring-4 focus:ring-green-200 group-hover:from-green-400 group-hover:to-blue-600 dark:text-white dark:focus:ring-green-800"
    >
      <a
        class="relative rounded-md bg-white px-5 py-2.5 transition-all duration-75 ease-in group-hover:bg-opacity-0 dark:bg-gray-900"
        href="Home"
      >
        Home
      </a>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
